docs/_guide/05-creating-directives.md

```diff
@@ -0,0 +1,221 @@
+
+Here's a trivial example of an asynchronous directive:
+
+```js
+const resolvePromise = directive((promise) => (part) => {
+  // This first setValue call is synchronous, so 
+  // doesn't need the commit
+  part.setValue("Waiting for promise to resolve.");
+
+  Promise.resolve(promise).then((resolvedValue) => {
+    part.setValue(resolvedValue);
+    part.commit();
+  });
+});
+```
+
+Here's an equally trivial example of the directive in use:
+
+```js
+const waitForIt = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    resolve("Promise is resolved.");
+  }, 1000);
+});
+
+const myTemplate = () => 
+   html`<div>${resolvePromise(waitForIt)}</div>`; 
+```
+
+Here, the rendered template shows "Waiting for promise to resolve," followed one second later by "Promise is resolved."
+
+## Repeating directives in content bindings
+
+Sometimes you want a directive to manage multiple nested parts. For example, a directive that renders a list of items (like `repeat`) might create a nested part for each item. Keeping separate parts lets you manipulate them efficiently: for example, you can change the value of a single part without re-rendering the entire list. 
+
+To create nested parts, you construct [`NodePart`](/api/classes/lit_html.nodepart.html) instances and associate them with specific locations in the DOM. The section of DOM controlled by a given `NodePart` needs to be delimited by static nodes that serve as markers. (lit-html usually uses comment nodes for these markers.)
+
+<img alt="Diagram showing a tree of DOM nodes and a NodePart object. The DOM tree consists of a parent node and seveal child nodes, with two of the child nodes identified as 'marker nodes.' The NodePart object has a startNode property, which points to the first marker node, and an endNode property, which points to the second marker node. Child nodes between the two marker nodes are identified as 'nodes managed by NodePart.'" src="/images/guides/node-part-markers.png" style="max-width: 515px;">
+
+As shown in the diagram, the nodes managed by the `NodePart` appear between its `startNode` and `endNode`. The following code creates and adds a new, nested part inside an existing part (the "container part"). 
+
+```js
+import {NodePart, createMarker} from 'lit-html';
+
+// Create a new part, passing in the render options from the original part
+const newPart = new NodePart(containerPart.options);
+
+// Create markers surrounding content managed by the new part
+const container = containerPart.startNode.parentNode;
+const startNode = createMarker();
+container.insertBefore(startNode, containerPart.endNode);
+container.insertBefore(createMarker(), containerPart.endNode);
+
+newPart.insertAfterNode(startNode);
+```
+
+If you don't need to manage the marker nodes yourself, you can use the [appendIntoPart](/api/classes/lit_html.nodepart.html#appendintopart) convenience method: 
```

Comment:
Suggestion: Maybe flip the order of these examples, showing `appendIntoPart` first. And then you can say something like "If you _do_ need to manage the marker nodes yourself, e.g. to insert the part into the container part somewhere other than at the end, you can create/insert/assign the marker nodes manually."  That might explicate _when_ you need to do the more complicated thing a little better?